### DISCUSSION TITLE
How to specify a timestamp range filter?

### DISCUSSION THREAD

**Discussion by omnine** [Kittyfisto/Tailviewer#347] [Q&A] [answered]
First of all, a very good log viewer!

Somehow, I cannot find an example on timestamp range filter.

`[2021-02-24 12:28:30]  [10.9.19.14]  [42616] [DEBUG] [CHttpModule::OnSendResponse] /mapi/emsmdb/?MailboxId=73dc4f95-873a-4f5c-8616-0aab5ae4b61a`

Can you please give me a hint for the above log sample? The first element is the timestamp.

Many thanks,

**Comment by Kittyfisto** [comment, marked as answer]
@omnine Thank you very much :)  

Regarding your question. I assume you want to filter the log entries by a time-range such as:
Only show log entries with a timestamp within a certain range [A, B]. In that case, this is partially possible, but only by date unfortunately, since I haven't come around to add / style a datetime input control.  

If you are looking for it, you can find it here:

1. Open the Filters Side Panel
![image](https://user-images.githubusercontent.com/11990827/110294951-ce99da00-7ff0-11eb-9f55-27484a74ad3e.png)

2. Click on "Select: Everything"
![image](https://user-images.githubusercontent.com/11990827/110294495-3bf93b00-7ff0-11eb-98a2-ad55309fa3de.png)

3a. Click on either "Quick" to have a simple filter (which is only ever relevant when looking at recent log files)
![image](https://user-images.githubusercontent.com/11990827/110295058-f25d2000-7ff0-11eb-8a81-409e06a6567c.png)

3b. Or on "Absolute" and select a start and end date:
![image](https://user-images.githubusercontent.com/11990827/110295242-2cc6bd00-7ff1-11eb-9017-7da3e4ac0a32.png)


The design is really unpolished and is in some need of an overhaul I'm afraid (apart from the ability to select time of day).  
Both of these things are on my todo list however.  


You may be able to workaround this with some creative substring or regexp quick filters:  
- "2021-02-24 12:" should filter all entries except those logged in that hour
- "2021-02-24 12:|2021-02-24 13:", set to regular expression should filter all entries except those logged in those two hours
![image](https://user-images.githubusercontent.com/11990827/110295863-ea51b000-7ff1-11eb-9700-1883670f5a2c.png)

Depending on the required complexity of the time-range you need to filter, you might be able to live with this workaround until I manage to create a better solution.

**Reply by omnine** [reply]
Thank you very much for the detailed explanation!
My log generally is on `minute` (even `second`) level, at the moment  I can live with the sub-string match (or regex).
